packages/graphics/src/utils/buildLine.ts

```diff
@@ -326,19 +326,56 @@ function buildNonNativeLine(graphicsData: GraphicsData, graphicsGeometry: Graphi
         const dx1 = x1 - x2;
         const dy1 = y2 - y1;
 
+        /* +ve if internal angle < 90 degree, -ve if internal angle > 90 degree. */
+        const dot = (dx0 * dx1) + (dy0 * dy1);
         /* +ve if internal angle counterclockwise, -ve if internal angle clockwise. */
         const cross = (dy0 * dx1) - (dy1 * dx0);
         const clockwise = (cross < 0);
 
-        /* Going nearly straight? */
-        if (Math.abs(cross) < 0.1)
+        /* Going nearly parallel? */
+        /* atan(0.001) ~= 0.001 rad ~= 0.057 degree */
+        if (Math.abs(cross) < 0.001 * Math.abs(dot))
         {
-            verts.push(
-                x1 - (perpx * innerWeight),
-                y1 - (perpy * innerWeight));
-            verts.push(
-                x1 + (perpx * outerWeight),
-                y1 + (perpy * outerWeight));
+            if (dot < 0)
+            {
+                // Straight line
+                verts.push(
+                    x1 - (perpx * innerWeight),
+                    y1 - (perpy * innerWeight));
+                verts.push(
+                    x1 + (perpx * outerWeight),
+                    y1 + (perpy * outerWeight));
+            }
+            else
+            {
+                // 180 degree corner
+                verts.push(
+                    x1 - (perpx * innerWeight),
+                    y1 - (perpy * innerWeight));
+                verts.push(
+                    x1 + (perpx * outerWeight),
+                    y1 + (perpy * outerWeight));
```

Comment:
Is there a reason you didn't pull this out of the if/else? These first two points seems the same to me.

```suggestion
           verts.push(
                x1 - (perpx * innerWeight),
                y1 - (perpy * innerWeight));
            verts.push(
                x1 + (perpx * outerWeight),
                y1 + (perpy * outerWeight));

            if (dot >= 0)
            {
```

Reply:
Sorry I messed up the whitespacing ☹️

Reply:
I'm fixing it :)